editor.planx.uk/src/pages/FlowEditor/lib/__tests__/navigation.test.ts

```diff
@@ -218,3 +218,48 @@ test("sectionStatuses() applies the NEW INFORMATION NEEDED status if a section w
     thirdSection: "CANNOT CONTINUE YET",
   });
 });
+
+test("sectionStatuses() corrects multiple in-progress statuses on reconciliation to reflect completed sections", () => {
+  setState({ flow: flowWithThreeSections });
+  initNavigationStore();
+
+  // Navigate forwards
+  record("firstSection", { auto: false });
+  record("firstQuestion", { answers: ["firstAnswer"] });
+  record("secondSection", { auto: false });
+  record("secondQuestion", { answers: ["secondAnswer"] });
+
+  // Mimic "reconciliation" by explicitly passing breadcrumbs instead of relying on cachedBreadcrumbs, assume no content has changed
+  const breadcrumbs = getState().breadcrumbs;
+  const statuses = sectionStatuses(breadcrumbs);
+
+  expect(statuses).toEqual({
+    firstSection: "COMPLETED",
+    secondSection: "COMPLETED",
+    thirdSection: "READY TO CONTINUE",
```

Comment:
"Mimicing reconciliation" is admittedly a bit strange for navigation store tests, but I think it's our best option until we have a full e2e with "resume" functionality. 

This test specifically fails against our prod code & sets each section status to `"READY TO CONTINUE"` which is confidence that my mocked sectionStatuses() arguments are accurate!